manifests/config.pp

```diff
@@ -34,5 +34,22 @@
     mode   => '0755',
   }
 
+  # NB: we define this here to avoid a dependency cycle. It is not a problem if
+  # this dir exists before the pulp RPMs are installed.
+  file { '/var/lib/pulp':
+    ensure  => directory,
+    owner   => 'apache',
+    group   => 'apache',
+    mode    => '0755',
+  }
+
+  file { $katello::repo_export_dir:
+    ensure  => directory,
+    owner   => $katello::user,
+    group   => $katello::group,
+    mode    => '0755',
+    # the default location requires this parent dir to exist
+    require => File['/var/lib/pulp']
```

Comment:
Since puppet 2.7 file resources auto-require all parents if they exist as puppet resources, so this should be redundant.